content/articles/building-a-multiclass-image-classification-model-using-mobilenet-v2-and-tensorflow/index.md

```diff
@@ -0,0 +1,363 @@
+
+#### Function for image preprocessing
+We create this function using this code:
+
+```python
+def preprocessing(image, label):
+  image /= 255.0
+
+  return tf.image.resize(image,[224,224]), tf.one_hot(label, 3)
+```
+Image normalization converts the output pixel value between 0 by 1. The `preprocessing` function performs `normalization` by dividing the `image` by 255. The function uses ` tf.image.resize` to resize the image to `224,224`. It is the image size that MobileNetV2 expects.
+
+The function also encodes the classes using `tf.one_hot(label, 3)`. It converts the classes (`Angular Leaf Spot`, `Bean Rust`, `healthy`), into numerical values (0, 1, 2)
+
+Let's now import libraries that we will use in image classification.
+
+### Libraries used
+Let's import the following:
+
+```python
+import tensorflow as tf
+import tensorflow_hub as hub
+import matplotlib.pylab as plt
+import numpy as np
+```
+
+**tensorflow**
+It is an open-source library for developing machine learning and deep learning models. It trains deep neural networks with input, hidden, and output layers. We will use TensorFlow to add custom layers to the pre-trained MobilenetV2. This will fine-tune the plant disease classification model and improve its performance.
+
+**tensorflow_hub**
+It is an open-source repository that contains pre-trained models for natural language processing tasks and image classification. We will download the pre-trained MobilenetV2 model from here.
+
+**matplotlib.pylab**
+We will the library to plot diagrams and visualization of our image dataset. It will show the model prediction results.
+
+**numpy**
+It will convert the image dataset into an array.
+
+Let's now download the pre-trained MobilenetV2 model which we will fine-tune later.
+
+### Pre-trained MobilenetV2
+We download the pre-trained MobilenetV2 using this code:
+
+```python
+mobilenet_v2 = "https://tfhub.dev/google/tf2-preview/mobilenet_v2/feature_vector/4"
+```
+As mentioned earlier, pre-trained MobilenetV2 is created based on the [convolutional neural network architecture](/engineering-education/basics-of-convolution-neural-networks/). This means it is made up of many convolution layers, pooling layers
+and a fully connected layer. 
+
+To use the pre-trained MobilenetV2 layers, use this code:
+
+```python
+mobile_net_layers = hub.KerasLayer(mobilenet_v2, input_shape=(224,224,3))
+```
+The code above will extract the pre-trained MobilenetV2 layers. It also specifies the size of the beans image we will feed the pre-trained MobilenetV2.
+
+`mobile_net_layers` will extract the unique characteristics and features from the beans images. We will then add our custom layers to the pre-trained MobilenetV2 using TensorFlow. It will produce a fine-tuned model that will classify the images into three classes.
+
+These layers are already trained. To ensure that TensorFlow does not retrain them, use this code:
```

Comment:
So, we are just using the pre-trained models to make predictions? If so, state that here. It's so not implicitly understandable.